portfolio/src/main/webapp/calendar.js

```diff
@@ -151,15 +152,180 @@ function collectAllEvents() {
     const startTime = new Date(eventCardBody.childNodes[1].innerText);
     const endTime = new Date(eventCardBody.childNodes[2].innerText);
     const event = new CalendarEvent(eventName, startTime, endTime);
-    return event; 
+    return event;
   });
-
+ 
   return allEventsOnPage;
 }
-
+ 
 /** Checks if two calendar events are identical. */
 function eventsEqual(eventA, eventB) {
   return (eventA.name === eventB.name) &&
       (eventA.startTime.getTime() === eventB.startTime.getTime()) &&
       (eventA.endTime.getTime() === eventB.endTime.getTime());
 }
+
+/** Retrives the date that the user has picked for the scheduling. */
+function getUserPickedDate() {
+  const userPickedDate = $('#date-picker').val().split('-');
+  const year = userPickedDate[0];
+  const month = userPickedDate[1];
+  const date = userPickedDate[2];
+ 
+  const pickedDate = new Date();
+  pickedDate.setFullYear(year);
+  pickedDate.setMonth(month - 1);  // month is zero-indexed.
+  pickedDate.setDate(date);
+ 
+  return pickedDate;
+}
+ 
+/** Sets the date picker to the current date as the default. */
+function setDatePickerToToday() {
+  const today = new Date();
+  const year = today.getFullYear();
+  const month = today.getMonth() + 1;  // month is zero-indexed
+  const date = today.getDate();
+  
+  const todayString = constructTodayString(year, month, date);
+  $('#date-picker').val(todayString);
+}
+
+/** 
+ * Parses the current day's date into a dash separated string. 
+ * The parameters are of type String. 
+ */
+function constructTodayString(year, month, date) {
```

Comment:
Consider clarifying in the method comment what year, month, date looks like. In particular, clarify whether month starts from 0 or 1.

Reply:
Updated.